File: frontend/src/types/attendance.ts
export type ActionType = 'CHECK_IN' | 'CHECK_OUT'
export type ResultType = 'SUCCESS' | 'FAILED'

export interface AttendanceRecord {
  id: number
  user_id?: number
  employee_id?: string
  name?: string
  action_type: ActionType | string
  confidence?: number
  snapshot_path?: string
  result: ResultType | string
  created_at: string
  device_id?: number
  device_name?: string
}

export interface AttendanceListResponse {
  items: AttendanceRecord[]
  total: number
  page: number
  page_size: number
}

export interface AttendanceStats {
  total_records: number
  success_count: number
  failed_count: number
  unique_users: number
  avg_confidence: number
}

export interface CheckInResponse {
  success: boolean
  action_type: string
  user?: {
    id: number
    employee_id: string
    name: string
  }
  confidence?: number
  message: string
  record_id?: number
}

File: backend/app/schemas/attendance.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum


class ActionType(str, Enum):
    CHECK_IN = "CHECK_IN"
    CHECK_OUT = "CHECK_OUT"


class ResultType(str, Enum):
    SUCCESS = "SUCCESS"
    FAILED = "FAILED"


class AttendanceBase(BaseModel):
    user_id: Optional[int] = None
    employee_id: Optional[str] = None
    name: Optional[str] = None
    action_type: ActionType
    confidence: Optional[float] = None
    result: ResultType


class AttendanceCreate(AttendanceBase):
    snapshot_path: Optional[str] = None


class AttendanceResponse(AttendanceBase):
    id: int
    device_id: Optional[int] = None
    device_name: Optional[str] = None
    snapshot_path: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class AttendanceListResponse(BaseModel):
    items: List[AttendanceResponse]
    total: int
    page: int
    page_size: int


class CheckInResponse(BaseModel):
    success: bool
    action_type: str
    user: Optional[dict] = None
    confidence: Optional[float] = None
    message: str
    record_id: Optional[int] = None


class AttendanceStats(BaseModel):
    total_records: int
    success_count: int
    failed_count: int
    unique_users: int
    avg_confidence: float
